client/allocrunner/networking_cni.go

```diff
@@ -133,6 +133,12 @@ func (c *cniNetworkConfigurator) Setup(ctx context.Context, alloc *structs.Alloc
 
 	addCustomCNIArgs(tg.Networks, cniArgs)
 
+	// Add NOMAD_* after custom args so it cannot be overridden.
+	cniArgs["NOMAD_NAMESPACE"] = alloc.Namespace
+	cniArgs["NOMAD_JOB_NAME"] = alloc.Job.Name
```

Comment:
I think this one should be ID, since Name is primarily ornamental, and may differ from the ID.

```suggestion
	cniArgs["NOMAD_JOB_ID"] = alloc.Job.ID
```

also just occurred to me - do we want to include `NOMAD_REGION` too? since it, along with Namespace and ID, uniquely identify the job globally.

Reply:
Thinking about it I am not sure, when doing network policies what would you expect it to match on? Maybe simply add both?

Reply:
In Nomad internally, only the job ID really "matters" -- the Name doesn't really affect anything, and it is not guaranteed to be unique like ID is (you can run two totally different jobs with the same name). So I would expect and encourage plugin authors to use job ID.

Users can still provide custom `cni_args` in the jobspec and include the name or anything else they'd like, or we can add it here later if there's demand for it, but personally I'd like to keep this default list pretty tight and tidy.

Reply:
Ah yes, good point!